src/main/java/net/logstash/logback/composite/JsonWritingUtils.java

```diff
@@ -171,4 +178,91 @@ public static boolean shouldWriteField(String fieldName) {
     }
 
+    /**
+     * Helper method to try to call appropriate write method for given
+     * untyped Object. At this point, no structural conversions should be done,
+     * only simple basic types are to be coerced as necessary.
+     *
+     * @param generator the {@link JsonGenerator} to produce JSON content
+     * @param value Value to write
+     *
+     * @throws IOException if there is either an underlying I/O problem or encoding
+     *    issue at format layer
+     */
+    public static void writeSimpleObject(JsonGenerator generator, Object value) throws IOException {
+        if (value == null) {
+            generator.writeNull();
+            return;
+        }
+        if (value instanceof String) {
+            generator.writeString((String) value);
+            return;
+        }
+        if (value instanceof Number) {
+            Number n = (Number) value;
+            if (n instanceof Integer) {
+                generator.writeNumber(n.intValue());
+                return;
+            }
+            if (n instanceof Long) {
+                generator.writeNumber(n.longValue());
+                return;
+            }
+            if (n instanceof Double) {
+                generator.writeNumber(n.doubleValue());
+                return;
+            }
+            if (n instanceof Float) {
+                generator.writeNumber(n.floatValue());
+                return;
+            }
+            if (n instanceof Short) {
+                generator.writeNumber(n.shortValue());
+                return;
+            }
+            if (n instanceof Byte) {
+                generator.writeNumber(n.byteValue());
+                return;
+            }
+            if (n instanceof BigInteger) {
+                generator.writeNumber((BigInteger) n);
+                return;
+            }
+            if (n instanceof BigDecimal) {
+                generator.writeNumber((BigDecimal) n);
+                return;
+            }
+            if (n instanceof AtomicInteger) {
+                generator.writeNumber(((AtomicInteger) n).get());
+                return;
+            }
+            if (n instanceof AtomicLong) {
+                generator.writeNumber(((AtomicLong) n).get());
+                return;
+            }
+        }
+        if (value instanceof byte[]) {
+            generator.writeBinary((byte[]) value);
+            return;
+        }
+        if (value instanceof Boolean) {
+            generator.writeBoolean((Boolean) value);
+            return;
+        }
+        if (value instanceof AtomicBoolean) {
+            generator.writeBoolean(((AtomicBoolean) value).get());
+            return;
+        }
+        if (value instanceof JsonNode) {
+            JsonNode node = (JsonNode) value;
+            
+            if (node instanceof NullNode) {
+                generator.writeNull();
+            } else {
+                node.serialize(generator, null); //FIXME
```

Comment:
FIXME?

Reply:
Ooops... I forgot this one.

Reply:
The FIXME was a reminder to add some comments with additional explanations about why the last argument is null. Comments are now added.